pycc/cclambda.py

```diff
@@ -120,7 +150,77 @@ def solve_lambda(self, e_conv=1e-7, r_conv=1e-7, maxiter=100, max_diis=8, start_
+                            Znf[n] += contract('de,def->f', l2[l,m], (t3_lmn - t3_lmn.swapaxes(0,2)))          
+                for i in range(no):  
+                    tmp = 0                 
+                    for m in range(no):
+                        tmp += contract('df,dfa->a', l2[i,m], Zmdfa[m])
+                        tmp += contract('af,f->a', L[i,m,v,v], Znf[m])  
+                        for n in range(no):                            
+                            tmp += contract('ad,d->a', l2[m,n], Zmndi[m,n,:,i])   
+                    Y1[i] += tmp
+                # end of t3l1
+                #l3l1+l3l2
+                Zbide = np.zeros((nv,no,nv,nv))
+                Zblad_1 = np.zeros((nv,no,nv,nv))
+                Zblad_2 = np.zeros((nv,no,nv,nv))
+                Zjlma = np.zeros((no,no,no,nv))
+                Zjlid_1 = np.zeros((no,no,no,nv))
+                Zjlid_2 = np.zeros((no,no,no,nv))
+                for i in range(no):
+                    for j in range(no):
+                        for k in range(no):
+                            l3_kij = l3_ijk(k, i, j, o, v, L, l1, l2, Fme, Wamef_cc3, Wmnie_cc3, F, contract, WithDenom=True)
+                            # l3l1_Z_build
+                            Zbide[:,i,:,:] += contract('bc,cde->bde', t2[j,k], l3_kij)
+                            Zblad_1[:,i,:,:] += contract('bc,cad->bad', t2[j,k], l3_kij)
+                            Zblad_2[:,i,:,:] += contract('bc,cda->bad', t2[j,k], l3_kij)  
+                            Zjlma[:,i,j,:] += contract('jbc,cab->ja', t2[:,k,:,:], l3_kij)
+                            Zjlid_1[:,i,j,:] += contract('jbc,cbd->jd', t2[:,k,:,:], l3_kij)
+                            Zjlid_2[:,i,j,:] += contract('jbc,cdb->jd', t2[:,k,:,:], l3_kij)
+                            # l3l2
+                            Y2[i,j] += contract('deb,eda->ab', l3_kij, Wabei_cc3[:,:,:,k]) 
+                            Y2[i] -= contract('dab,ld->lab', l3_kij, Wmbij_cc3[:,:,j,k])
+                # l3l1
+                for i in range(no):                   
+                    tmp = contract('bde,deab->a', Zbide[:,i,:,:], Wabef_cc3)
+                    for j in range(no):
+                        for l in range(no):
+                            for m in range(no):                               
+                                tmp += Zjlma[j,l,m] * Wmnij_cc3[i,j,l,m]
+                    for j in range(no):
+                        for l in range(no):
+                            tmp -= contract('d,da->a', Zjlid_1[j,l,i,:], Wmbje_cc3[j,:,l,:])
+                            tmp -= contract('d,da->a', Zjlid_2[j,l,i,:], Wmbej_cc3[j,:,:,l])
+                    for l in range(no):
+                        tmp -= contract('bad,db->a', Zblad_1[:,l,:,:], Wmbje_cc3[i,:,l,:])
+                        tmp -= contract('bad,db->a', Zblad_2[:,l,:,:], Wmbej_cc3[i,:,:,l])   
+                    Y1[i] += tmp
+                # end l3l1+l3l2
```

Comment:
Same question about re-factorization as above. I think these loops can be better optimized.

Reply:
Thanks for pointing that out. I will making modifications accordingly.